src/Database/Schema/SqlserverSchema.php

```diff
@@ -371,12 +371,26 @@ public function columnSql(TableSchema $schema, $name)
         }
 
         if ($data['type'] === TableSchema::TYPE_BINARY) {
-            $out .= ' VARBINARY';
-
-            if ($data['length'] !== TableSchema::LENGTH_TINY) {
-                $out .= '(MAX)';
+            if (isset($data['length']) && $data['length'] === 1) {
+                $out .= ' BINARY(1)';
             } else {
-                $out .= sprintf('(%s)', TableSchema::LENGTH_TINY);
+                $out .= ' VARBINARY';
+
+                if (empty($data['length']) || in_array(
+                    $data['length'],
+                    [TableSchema::LENGTH_MEDIUM, TableSchema::LENGTH_LONG],
+                    true
+                )) {
+                    $out .= '(MAX)';
+                } else {
```

Comment:
This first of condition could be folded into the if ladder inside of it avoiding a layer of nesting.

Reply:
Done.